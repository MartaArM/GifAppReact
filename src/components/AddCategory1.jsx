import { useState } from "react"

// En este componente se pasa la funcion setCategories desde el principal hacia este.


export const AddCategory1 = ( {setCategories}) /*Funcion que cambia las categorias en el componente principal */ => {

    const[valueInput, setValueInput] = useState(""); // Valor del input

    const inputChange = (event) => { //Escribo en el input
        setValueInput(event.target.value); // Coger el valor del input cuando cambia
    }

    const submitForm = (event) => { // Cuando pulso la tecla enter en el input
        event.preventDefault(); //Se utiliza para que no se recargue la pÃ¡gina cuando le das enter

        if (valueInput.trim().length>1) { //Si lo que inserto tiene longitud mayor a 1
            setCategories((categories) => [valueInput, ...categories]);
            setValueInput(''); // Limpiamos el valor para que en el input desaparezca
        }
    }

    return (
        <form onSubmit = {submitForm}>
            <input
            type="text"
            placeholder="Buscar..."
            value={valueInput}
            onChange = {inputChange} // No hace falta pasarle el event
            />
        </form>
    )
}