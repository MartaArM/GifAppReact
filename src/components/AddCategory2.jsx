import { useState } from "react"

// En este componente se pasa la funcion setCategories desde el principal hacia este.


export const AddCategory2 = ( {newCategory}) /*Funcion que cambia las categorias en el componente principal */ => {

    const[valueInput, setValueInput] = useState(""); // Valor del input

    const inputChange = (event) => { //Escribo en el input
        setValueInput(event.target.value); // Coger el valor del input cuando cambia
    }

    const submitForm = (event) => { // Cuando pulso la tecla enter en el input
        const valueTrim = valueInput.trim();
        event.preventDefault(); //Se utiliza para que no se recargue la pÃ¡gina cuando le das enter

        if (valueTrim.length>1) { //Si lo que inserto tiene longitud mayor a 1
            newCategory(valueTrim); // Pasamos el valor a la funcion principal en vez de cambiarlo aqui
            setValueInput(''); // Limpiamos el valor para que en el input desaparezca
        }
    }

    return (
        <form onSubmit = {submitForm}>
            <input
            type="text"
            placeholder="Buscar..."
            value={valueInput}
            onChange = {inputChange} // No hace falta pasarle el event
            />
        </form>
    )
}