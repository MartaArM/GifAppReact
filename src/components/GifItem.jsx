import PropTypes from 'prop-types';

// ELemento para mostrar los gifs
export const GifItem = ({title ,url, id})  => {
    return (
        <div className="card">
            <img src={url}></img>
            <p>{title}</p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired, 
    url: PropTypes.string.isRequired
}