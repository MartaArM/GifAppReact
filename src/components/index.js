// CReamos un archivo de barril para importar elementos de la carpeta components

export * from './AddCategory2';
export * from './GifGrid';
export * from './GifItem';