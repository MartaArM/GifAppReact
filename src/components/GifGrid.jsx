import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';



// ELemento para mostrar los gifs
export const GifGrid = ( {category})  => {
    
    
    // Vamos a usar un custom hook
    const {gifs, isLoading} = useFetchGifs(category);

    
    {
        isLoading ? (<h2> Cargando.. </h2>) : null // Muestra un mensaje de cargando mientras que se cargan las imagenes
    }


    return (
        <>
            <h3>{ category }</h3>
            <div className="card-grid">
            {
                gifs.map( (element) =>  ( // Por cada categor√≠a creamos una lista de gifs
                    <GifItem key={element.id} {...element}/> // Aqui se muestran los GIF
                ) )
            }

            </div>
        </>
    )
}