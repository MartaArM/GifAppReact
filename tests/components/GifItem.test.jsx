import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en GifItem', () => {
    const title = 'Mi gif';
    const url = "https://media0.giphy.com/media/26xBwdIuRJiAIqHwA/200.gif?cid=fe9f259bmez0qjhl3kqc1uu4xiqmvc5byalxqm4mgaxjnows&rid=200.gif&ct=g";
    test('Snapshot', ()=> {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect( container).toMatchSnapshot();
    });
});