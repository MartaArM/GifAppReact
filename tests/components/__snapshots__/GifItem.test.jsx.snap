// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      src="https://media0.giphy.com/media/26xBwdIuRJiAIqHwA/200.gif?cid=fe9f259bmez0qjhl3kqc1uu4xiqmvc5byalxqm4mgaxjnows&rid=200.gif&ct=g"
    />
    <p>
      Mi gif
    </p>
  </div>
</div>
`;
